lib/emitter.ts

```diff
@@ -4,7 +4,8 @@ export class EventEmitter {
   private events: { [key: string]: Listener } = {};
 
   public on(event: string, listener: Listener): void {
-    if ({}.toString.call(listener) !== '[object Function]')
+    const s: string = {}.toString.call(listener);
+    if (s !== '[object Function]' && s !== '[object AsyncFunction]')
```

Comment:
Can you change that to `typeof listener === 'function'`

Reply:
Thanks to @GrosSacASac , I've updated the code.